readthedocs/api/v3/views.py

```diff
@@ -161,6 +178,33 @@ def get_view_description(self, *args, **kwargs):  # pylint: disable=arguments-di
                 return mark_safe(description.format(project_slug=project.slug))
         return description
 
+    def create(self, request, *args, **kwargs):
+        """
+        Override method to importing a Project.
+
+        * Save the Project object
+        * Assign the user from the request as owner
+        * Sent project_import signal
+        * Trigger an initial Build
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        project = serializer.save()
+        headers = self.get_success_headers(serializer.data)
+
+        # TODO: these lines need to be adapted for Corporate
+        project.users.add(request.user)
```

Comment:
Where does this happen in the normal import workflow? I can't find it in the code.

Reply:
Ah, was on the subclass: https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/projects/forms.py#L50-L53

Reply:
So perhaps we should be inheriting form this form for the Form we're using?

Reply:
I'm not sure I like to inherit a Serializer from a Wizard Form. Although, maybe these common methods could be extracted into a Mixin that it's usage by both. Actually, in that case, there is nothing to change in corporate since it's already done the override.

Reply:
This involves a bigger refactor than I thought. All the process to import a project is very tied to forms methods. `ProjectBasicForm.save` links the remote repository.

`ImportWizardView` manipulates the Basic and the Extra form to make it all fit in one project before saving it plus triggering a new build (`done` method in the form)